### PR TITLE
update resource handling documentation

### PR DESCRIPTION
**Pull request by SimonTheLeg**
Signed-off-by: Simon Bein <simontheleg@gmail.com>

**What does this PR do / Why do we need it**:

Update documentation on resource handling

**Does this PR close any issues?**:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: docs/resource-handling.md [hunk 1]
```diff
@@ -121,14 +113,14 @@ This avoids the need to call the creator function twice (1st time to get the obj
...
 ```go
-func MyWonderfulSecretCreator() resources.NamedSecretCreatorGetter {
-	return func() (string, resources.SecretCreator) {
-		return "my-name", func(existing *corev1.Secret) (*corev1.Secret, error) {
-			existing.Name = "wonderful-secret"
+func MyWonderfulSecretCreator() reconciling.NamedSecretCreatorGetter {
+	return func() (string, reconciling.SecretCreator) {
+		return "my-wonderful-secret", func(existing *corev1.Secret) (*corev1.Secret, error) {
+
+			// We don't need to set the object's name, as that's being done by the typed Reconcile function.
```
Comment (xrstf) [inline docs/resource-handling.md:123]: name _or namespace_

**Review by xrstf — CHANGES_REQUESTED** [review]
One minor improvement left

File: docs/resource-handling.md (hunk 1 above)
Comment (SimonTheLeg) [reply docs/resource-handling.md:123]: As far as I understand it, "name". Afaik the namespace should not be set in the CreatorGetter, but rather passed directly to the corresponding typed Reconcile function.

That being said, I have now added a sentence to this paragraph to make this more clear. ptal

**Review by xrstf — APPROVED** [review]
/approve

**Pull request merged**